forte/utils/payload_factory.py

```diff
@@ -0,0 +1,159 @@
+from abc import abstractmethod
+from multiprocessing.sharedctypes import Value
+from typing import Callable
+import numpy as np
+from forte.data.ontology.top import Meta
+
+
+class PayloadFactory:
+    """
+    A class that handles the creation of payloads. It can register meta data and check its validity.
+    """
+
+    def __init__(self):
+        self.valid_meta = (
+            {}
+        )  # map from meta data name to function that handles this meta data
+
+    def register(self, meta: Meta):
+        """
+        A function that registers a meta data type into the factory.
+
+        Args:
+            meta_name: a Generic object that is used to register a Payload meta data type.
+        """
+        if meta.source_type not in ("web", "local"):
+            raise ValueError("Meta data source must be either 'web' or 'local'")
```

Comment:
hmm, I don't think you want to control these things here. Basically you are making a lot of assumption of the data source